Sources/MapboxNavigation/NavigationMapView+Annotations.swift

```diff
@@ -0,0 +1,488 @@
+            "imageName": .string(imageName),
+            "sortOrder": .number(Double(isSelected ? index : -index)),
+            "routeIndex": .number(Double(index))
+        ]
+        
+        return feature
+    }
+    
+    /**
+     Generate the text for the label to be shown on screen. It will include estimated duration
+     and info on Tolls, if applicable.
+     */
+    private func annotationLabelForRoute(_ route: Route, tolls: Bool) -> String {
+        let eta = DateComponentsFormatter.shortDateComponentsFormatter.string(from: route.expectedTravelTime) ?? ""
+        
+        return tollAnnotationForLabel(on: route, tolls: tolls, label: eta)
+    }
+    
+    private func tollAnnotationForLabel(on route: Route?, tolls: Bool, label: String) -> String {
+        var labelWithTolls = label
+        let hasTolls = (route?.tollIntersections?.count ?? 0) > 0
+        if hasTolls {
+            labelWithTolls += "\n" + NSLocalizedString("ROUTE_HAS_TOLLS", bundle: .mapboxNavigation, value: "Tolls", comment: "This route does have tolls")
+            if let symbol = Locale.current.currencySymbol {
+                labelWithTolls += " " + symbol
+            }
+        } else if tolls {
+            // If one of the routes has tolls, but this one does not then it needs to explicitly say that it has no tolls
+            // If no routes have tolls at all then we can omit this portion of the string.
+            labelWithTolls += "\n" + NSLocalizedString("ROUTE_HAS_NO_TOLLS", bundle: .mapboxNavigation, value: "No Tolls", comment: "This route does not have tolls")
+        }
+        
+        return labelWithTolls
+    }
+    
+    /**
+     Generate the text for the label to be shown on screen. It will include estimated duration delta relative to the main route
+     and info on Tolls, if applicable.
+     */
+    private func annotationLabelForAlternativeRoute(_ alternativeRoute: AlternativeRoute, tolls: Bool) -> String {
+        let timeDelta = DateComponentsFormatter.travelTimeString(alternativeRoute.expectedTravelTimeDelta,
+                                                                 signed: true,
+                                                                 unitStyle: nil)
+        
+        return tollAnnotationForLabel(on: alternativeRoute.indexedRouteResponse.currentRoute,
+                                      tolls: tolls,
+                                      label: timeDelta)
+    }
+    
+    // MARK: Intersection Signals Annotations
+    
+    /**
+     Updates the image assets in the map style for the route intersection signals.
+     */
+    func updateIntersectionSymbolImages() throws {
+        let style = mapView.mapboxMap.style
+        
+        if style.image(withId: ImageIdentifier.trafficSignalDay) == nil,
+           let trafficSignlaDay =  Bundle.mapboxNavigation.image(named: "TrafficSignalDay") {
+            try style.addImage(trafficSignlaDay, id: ImageIdentifier.trafficSignalDay)
+        }
```

Comment:
It would be awesome if the current `Style` subclass could define these images right alongside `Style.mapStyle`.